substrate/client/cli/src/commands/generate_node_key.rs

```diff
@@ -36,18 +37,33 @@ use std::{
 pub struct GenerateNodeKeyCmd {
 	/// Name of file to save secret key to.
 	/// If not given, the secret key is printed to stdout.
-	#[arg(long)]
+	#[arg(long, conflicts_with_all = ["base_path", "default_base_path", "chain"])]
 	file: Option<PathBuf>,
 
 	/// The output is in raw binary format.
 	/// If not given, the output is written as an hex encoded string.
 	#[arg(long)]
 	bin: bool,
+
+	/// Specify the chain specification.
+	///
+	/// It can be any of the predefined chains like dev, local, staging, polkadot, kusama.
+	#[arg(long, value_name = "CHAIN_SPEC")]
+	pub chain: Option<String>,
+	/// A directory where the key should be saved. If a key already
+	/// exists in the directory, it won't be overwritten.
+	#[arg(long, conflicts_with_all = ["file", "default_base_path"])]
+	base_path: Option<PathBuf>,
+
+	/// Save the key in the default directory. If a key already
+	/// exists in the directory, it won't be overwritten.
+	#[arg(long, conflicts_with_all = ["base_path", "file"])]
+	default_base_path: bool,
```

Comment:
Maybe we should just remove this and `chain` completely. IMO, if you run a validator you should not use the default base path and even if you use it, you could still use the `unsafe-force-generate-node-key` CLI command.

Reply:
Care to develop here? Why shouldn't a default validator use the default-base-path what issue can arise from that  ? I thought most  of them actually do it

Reply:
IDK :P These paths are "magic" and you will first need to do some digging to find them. Especially if you want to mount this folder in docker or whatever. I would personally find it easier to pass some explicit path, but maybe I'm weird xD

Nevertheless, if you want to keep this, we should introduce a new `GenerateNodeKeyCmd` that should be used for "nodes". If you use the dedicated `subkey` binary it doesn't make sense to have the `chain` parameter etc. Does this makes sense?

Reply:
Yes, I agree `chain` parameter doesn't make sense for the subkey binary, however it needs to be present for `node` because even when you run polkadot with a specific `base-path` we append to it the chain_id to obtain the configuration directory.

Thinking of  going the route of introducing a new GenerateNodeKeyCmd that should be used for nodes, would that work for you ?

Reply:
Done, let me know if it is better now.